src/sidebar/services/frame-sync.ts

```diff
@@ -267,6 +282,33 @@ export class FrameSyncService {
           this._updateAnchorStatus(anchorImmediately, 'anchored');
         }
 
+        // For added annotations that still need to anchor, re-check them after
+        // ANCHORING_TIMEOUT has elapsed. Any added annotations not anchored
+        // by that time should be marked as having an anchor timeout.
+        const anchoringIds = added
+          .filter(ann => isWaitingToAnchor(ann))
+          .map(ann => ann.id);
+
+        const ANCHORING_TIMEOUT = 500;
+
+        if (anchoringIds.length > 0) {
+          const anchoringTimeout = setTimeout(() => {
+            const annsStillAnchoring = anchoringIds
+              .map(id => (id ? this._store.findAnnotationByID(id) : null))
```

Comment:
This could be simplified if a store method was added to find annotations by tag. That functionality exists internally within the annotations module, it just needs to be exposed.